x/upgrade/tally_test.go

```diff
@@ -0,0 +1,249 @@
+		ValidatorAddress: testutil.ValAddrs[0].String(),
+		Version:          3,
+	})
+	require.Error(t, err)
+
+	_, err = upgradeKeeper.SignalVersion(goCtx, &types.MsgSignalVersion{
+		ValidatorAddress: testutil.ValAddrs[0].String(),
+		Version:          2,
+	})
+	require.NoError(t, err)
+
+	res, err := upgradeKeeper.VersionTally(goCtx, &types.QueryVersionTallyRequest{
+		Version: 2,
+	})
+	require.NoError(t, err)
+	require.EqualValues(t, 30, res.VotingPower)
+	require.EqualValues(t, 100, res.Threshold)
+	require.EqualValues(t, 120, res.TotalVotingPower)
+
+	_, err = upgradeKeeper.SignalVersion(goCtx, &types.MsgSignalVersion{
+		ValidatorAddress: testutil.ValAddrs[2].String(),
+		Version:          2,
+	})
+	require.NoError(t, err)
+
+	res, err = upgradeKeeper.VersionTally(goCtx, &types.QueryVersionTallyRequest{
+		Version: 2,
+	})
+	require.NoError(t, err)
+	require.EqualValues(t, 80, res.VotingPower)
+	require.EqualValues(t, 100, res.Threshold)
+	require.EqualValues(t, 120, res.TotalVotingPower)
+
+	upgradeKeeper.EndBlock(ctx)
+	shouldUpgrade, version := upgradeKeeper.ShouldUpgrade()
+	require.False(t, shouldUpgrade)
+	require.Equal(t, uint64(0), version)
+
+	// modify the quorum so we are right on the boundary 80/120 = 2/3
+	newParams := types.DefaultParams()
+	newParams.SignalQuorum = types.MinSignalQuorum
+	upgradeKeeper.SetParams(ctx, newParams)
+
+	upgradeKeeper.EndBlock(ctx)
+	shouldUpgrade, version = upgradeKeeper.ShouldUpgrade()
+	require.False(t, shouldUpgrade)
+	require.Equal(t, uint64(0), version)
+	require.EqualValues(t, 80, upgradeKeeper.GetVotingPowerThreshold(ctx).Int64())
+
+	// we now have 81/120 = 0.675 > 2/3
+	_, err = upgradeKeeper.SignalVersion(goCtx, &types.MsgSignalVersion{
+		ValidatorAddress: testutil.ValAddrs[1].String(),
+		Version:          2,
+	})
+	require.NoError(t, err)
+
+	upgradeKeeper.EndBlock(ctx)
+	shouldUpgrade, version = upgradeKeeper.ShouldUpgrade()
+	require.True(t, shouldUpgrade)
+	require.Equal(t, uint64(2), version)
+	// update the version to 2
+	ctx = ctx.WithBlockHeader(tmproto.Header{
+		Version: tmversion.Consensus{
+			Block: 1,
+			App:   2,
+		},
+	})
+	goCtx = sdk.WrapSDKContext(ctx)
+
+	_, err = upgradeKeeper.SignalVersion(goCtx, &types.MsgSignalVersion{
+		ValidatorAddress: testutil.ValAddrs[0].String(),
+		Version:          3,
+	})
+	require.NoError(t, err)
+
+	res, err = upgradeKeeper.VersionTally(goCtx, &types.QueryVersionTallyRequest{
+		Version: 2,
+	})
+	require.NoError(t, err)
+	require.EqualValues(t, 51, res.VotingPower)
+	require.EqualValues(t, 80, res.Threshold)
+	require.EqualValues(t, 120, res.TotalVotingPower)
+
+	// remove one of the validators from the set
+	delete(mockStakingKeeper.validators, testutil.ValAddrs[2].String())
+	mockStakingKeeper.totalVotingPower -= 50
+
+	res, err = upgradeKeeper.VersionTally(goCtx, &types.QueryVersionTallyRequest{
+		Version: 2,
+	})
+	require.NoError(t, err)
+	require.EqualValues(t, 1, res.VotingPower)
+	require.EqualValues(t, 47, res.Threshold)
```

Comment:
[no change needed] this makes me realize that threshold got rounded up from `46.66666667`. I think it's unlikely but should the threshold always round up instead of rounding to the nearest int? 

Update: commented above on a more relevant line where Threshold is computed

Reply:
So I think it's okay to use `RoundInt()` instead of `Ceil()` (i.e round to the nearest integer instead of the next highest integer) because the comparison logic is strictly greater than this threshold rather than greater than or equal to. i.e. if we rounded down to 46, it would still take 47 in voting power to advance. 

Alternatively, I do `Ceil` and then invoke >= for the comparison logic